Automattic/wp-calypso
# Goals Capture: Implement 'Skip to dashboard' handler

#### Proposed Changes

* Re-write `Skip to Dashboard` button label to `Skip to dashboard`
* Reset `intent` (set it to empty string), stored in redux onboard store, every time `goals` step is opened
* Do not resolve and store intent on every goals selected change
* Resolve and store intent only on submit, i.e. when `Continue` button is clicked
* On pressing `Skip to dashboard` button on goals capture screen: 
  * user is redirected to calypso dashboard
  * call to `POST site-intent` api is made with `{ "site_intent": "" }`
  * if any goals on the screen were selected they are also submitted to `POST site-goals`

#### Testing Instructions



1. Navigate to `http://calypso.localhost:3000/setup/goals?siteSlug=<your-site>` 
2. Press the `Skip to dashboard` button:
![Screenshot 2022-06-22 at 10 37 58](https://user-images.githubusercontent.com/2019970/174986433-749a224b-d407-497a-a95e-5d15c1858031.png)
3. You should be redirected to the sites calypso dashboard:
![Screenshot 2022-06-22 at 10 38 19](https://user-images.githubusercontent.com/2019970/174986438-674f8897-fd3f-4c5b-8442-65923a110bb1.png)
4. Ensure the `POST site-intent` and `POST site-goals` api calls were made when the redirection happened.
![Screenshot 2022-06-22 at 10 38 50](https://user-images.githubusercontent.com/2019970/174986445-e6e4a3d8-a1e0-4632-b7e6-27170646a52e.png)
![Screenshot 2022-06-22 at 10 39 03](https://user-images.githubusercontent.com/2019970/174986446-29813261-401a-4814-8bae-b614af158dc4.png)
